### ISSUE TITLE
Saving Optimizer and Scheduler state at checkpoints

### ISSUE THREAD

**Issue by GeorgeSanchezTR** [UKPLab/sentence-transformers#1086]
Using the model.fit method and supplying the checkpoint_save_path only saves the model but not the optimizer and scheduler states. Continuing training from a saved checkpoint means that the learning rate needs to be set again.
Is there a way to save the optimizer and scheduler state inside model.fit so that when we try to continue training it will use the checkpoints learning rate - much like Trainer.py in hugginface. 

https://github.com/UKPLab/sentence-transformers/blob/fb7b7a3044f21de5b54f466f732baa48ba1ae4ba/sentence_transformers/SentenceTransformer.py#L723
save

**Comment by nreimers** [comment]
Currently it is not implemented. Will add it to the next release
